### PR TITLE
[SofaCUDA] Fix cuda with latest API

### PR DESCRIPTION
**Pull request by courtecuisse**
Fix cuda compilation (problems with float/double in cuda)
______________________________________________________

This PR: 
- [ ] builds with SUCCESS for all platforms on the CI.
- [ ] does not generate new warnings.
- [ ] does not generate new unit test failures.
- [ ] does not generate new scene test failures.
- [ ] does not break API compatibility.
- [ ] is more than 1 week old (or has fast-merge label).

**Reviewers will merge only if all these checks are true.**

### CODE REVIEW

File: applications/plugins/SofaCUDA/sofa/gpu/cuda/CudaIdentityMapping.cpp [hunk 1]
```diff
@@ -54,7 +54,7 @@ template class  IdentityMapping< CudaVec3dTypes, CudaVec3fTypes>;
 template class  IdentityMapping< CudaVec3dTypes, CudaVec3dTypes>;
 template class  IdentityMapping< CudaVec3dTypes, Vec3fTypes>;
 template class  IdentityMapping< CudaVec3dTypes, Vec3dTypes>;
-template class  IdentityMapping< Vec3Types, CudaVec3Types>;
+//template class  IdentityMapping< Vec3Types, CudaVec3Types>;
```
Comment (damienmarchal) [inline applications/plugins/SofaCUDA/sofa/gpu/cuda/CudaIdentityMapping.cpp]: We shouldn't merge commented code. Juste remove it.

File: SofaKernel/modules/SofaRigid/JointSpringForceField.h [hunk 2]
```diff
@@ -37,6 +37,16 @@ namespace component
 namespace interactionforcefield
 {
 
+//This class is specialized with cudatypes
+template<class DataTypes>
```
Comment (damienmarchal) [inline SofaKernel/modules/SofaRigid/JointSpringForceField.h:41]: For curiousity I searched for the specialization but didn't find it. Is it in a private plugin ?

**Comment by damienmarchal** [comment]
Hi Adrien,

Thanks for the PR. 
@guparan, to me apart from the merged commented code this looks OK. 

Damien.

File: applications/plugins/SofaCUDA/sofa/gpu/cuda/CudaDiagonalMass.cpp [hunk 3]
```diff
@@ -24,6 +24,11 @@
 #include <sofa/core/behavior/Mass.inl>
 #include <sofa/core/behavior/ForceField.inl>
 #include <sofa/core/ObjectFactory.h>
+#include <SofaBaseTopology/EdgeSetGeometryAlgorithms.inl>
```
Comment (epernod) [inline applications/plugins/SofaCUDA/sofa/gpu/cuda/CudaDiagonalMass.cpp:27]: Include .inl or .h . I'm not expert, what is the best practice?

Comment (fredroy) [reply applications/plugins/SofaCUDA/sofa/gpu/cuda/CudaDiagonalMass.cpp:27]: The inl is needed for the implementation of the templated functions, as we instanciate classes with float (and those has been removed from the core of Sofa)

Comment (damienmarchal) [reply applications/plugins/SofaCUDA/sofa/gpu/cuda/CudaDiagonalMass.cpp:27]: The inl are needed each time your have new template instantiations (in this case the float ones).

File: applications/plugins/SofaCUDA/sofa/gpu/cuda/CudaIdentityMapping.cpp (hunk 1 above)
Comment (guparan) [reply applications/plugins/SofaCUDA/sofa/gpu/cuda/CudaIdentityMapping.cpp]: Done.

**Comment by damienmarchal** [comment]
Merged, thanks for the PR Hadrien.

**Pull request merged**